plugin/gluoncv_backend.py

```diff
@@ -0,0 +1,439 @@
+                         .with_model_uri(base_name) \
+                         .build()
+        return (new_config, [local_path])
+
+    def load_bundle_files(self, bundle_dir):
+        if not self.model_uri:
+            raise rv.ConfigError('model_uri is not set.')
+        local_model_uri = os.path.join(bundle_dir, self.model_uri)
+        return self.to_builder() \
+                   .with_model_uri(local_model_uri) \
+                   .build()
+
+
+class GluonCVBackendConfigBuilder(BackendConfigBuilder):
+    def __init__(self, prev=None):
+        config = {}
+        if prev:
+            config = {
+                'train_options': prev.train_options,
+                'debug': prev.debug,
+                'training_data_uri': prev.training_data_uri,
+                'training_output_uri': prev.training_output_uri,
+                'model_uri': prev.model_uri,
+                'pretrained_model_uri': prev.pretrained_model_uri
+            }
+        self.require_task = prev is None
+        super().__init__(GLUONCV_BACKEND, GluonCVBackendConfig, config)
+
+    def from_proto(self, msg):
+        b = super().from_proto(msg)
+        conf = msg.custom_config
+        b.require_task = None
+
+        b = b.with_train_options(
+            sync_interval=conf['sync_interval'],
+            do_monitoring=conf['do_monitoring'],
+            replace_model=conf['replace_model'],
+        )
+        b = b.with_training_data_uri(conf['training_data_uri'])
+        b = b.with_training_output_uri(conf['training_output_uri'])
+        b = b.with_model_uri(conf['model_uri'])
+
+        return b
+
+    def _applicable_tasks(self):
+        return [rv.CHIP_CLASSIFICATION]
+
+    def _process_task(self):
+        return self
+
+    def with_config(self,
+                    config_mod,
+                    ignore_missing_keys=False,
+                    set_missing_keys=False):
+        """Given a dict, modify the tensorflow pipeline configuration
+           such that keys that are found recursively in the configuration
+           are replaced with those values. TODO: better explination.
+        """
+        b = deepcopy(self)
+        b.config_mods.append((config_mod, ignore_missing_keys,
+                              set_missing_keys))
+        return b
+
+    def with_train_options(self,
+                           sync_interval=600,
+                           do_monitoring=True,
+                           replace_model=False):
+        """Sets the train options for this backend.
+
+           Args:
+              sync_interval: How often to sync output of training to
+                             the cloud (in seconds).
+
+              replace_model: Replace the model checkpoint if exists.
+                             If false, this will continue training from
+                             checkpoing if exists, if the backend allows
+                             for this.
+        """
+        b = deepcopy(self)
```

Comment:
~~You should get rid of this method since none of these options are actually implemented. It could be confusing otherwise.~~ Since the work is already done to get these options into the gluon_cv training routine we should just leave it as is, but raise an exception that says these option aren't implemented yet or something. I would normally say to remove dead code, but it seems like this will be useful in the future and I don't want to waste time adding the stuff back at that time.